### PR TITLE
More arm intrinsics (RFC: additional tests)

### PR DESCRIPTION
**Pull request by sunnygleason**
WIP...

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,5 @@ Cargo.lock
 .*.swp
 target
 tags
+crates/stdarch-gen/aarch64.rs
+crates/stdarch-gen/arm.rs
```
Comment (bjorn3) [inline .gitignore]: Missing trailing newline

**Comment by Licenser** [comment]
@sunnygleason you are so much more patient then I am! Huge respect mate!

File: crates/core_arch/src/aarch64/neon/generated.rs [hunk 2]
```diff
@@ -297,292 +297,320 @@ mod test {
     use std::mem::transmute;
     use stdarch_test::simd_test;
 
+    macro_rules! verify_fn {
```
Comment (Licenser) [inline crates/core_arch/src/aarch64/neon/generated.rs]: Additional tests would need to live in a different file `mod.rs` for example? This file, being generated will be completely overwritten on updates.

**Comment by sunnygleason** [comment]
@Licenser happy to do it, yay teamwork!

**Comment by sunnygleason** [comment]
OK I think the aarch64 tests should be all set here. I'll work on the arm tests later this evening!

**Comment by sunnygleason** [comment]
@Licenser one thing I saw, not sure if you're getting it too? Figured
maybe you'd recognize how to fix these...

```
---- verify_all_signatures stdout ----
failed to verify `vuqadd_s8`
  * mismatched arguments
failed to verify `vuqaddq_s8`
  * mismatched arguments
failed to verify `vuqadd_s16`
  * mismatched arguments
failed to verify `vuqaddq_s16`
  * mismatched arguments
failed to verify `vuqadd_s32`
  * mismatched arguments
failed to verify `vuqaddq_s32`
  * mismatched arguments
failed to verify `vreinterpretq_u8_s8`
  * mismatched arguments
thread 'verify_all_signatures' panicked at 'assertion failed: all_valid', crates/stdarch-verify/tests/arm.rs:236:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Comment by Licenser** [comment]
Good catch on the vuqadd intrinsics, they require different first and second types that currently won't get generated so simple solution - I remove them for now.

**Comment by sunnygleason** [comment]
@Licenser @bjorn3 I've added a substantial number of new tests to the arm/mod.rs file,
hopefully this looks appealing!

If you have time to review the arm side, I'd be most appreciative. I'll be applying the tests
to the aarch64 side in the meantime. Thanks everyone again for their kind help, hopefully
we're close to getting this across the finish line.

**Review by Licenser — APPROVED** [review]
Awesome work!

**Comment by Licenser** [comment]
ready merge whenever you think it's good :)

**Review by Licenser — APPROVED** [review]

**Comment by sunnygleason** [comment]
@Licenser ok I think this should be good to go! (GH says I don't have write access, so I think you need to merge the PR). Thank you again!

**Comment by Licenser** [comment]
Wow awesome! Thank you!

**Pull request merged**